pkg/trace/remoteconfighandler/remote_config_handler.go

```diff
@@ -43,9 +52,25 @@ type RemoteConfigHandler struct {
 	agentConfig                   *config.AgentConfig
 	configState                   *state.AgentConfigState
 	configSetEndpointFormatString string
+	concentrator                  concentrator
 }
 
-func New(conf *config.AgentConfig, prioritySampler prioritySampler, rareSampler rareSampler, errorsSampler errorsSampler) *RemoteConfigHandler {
+func mapsAreEqual(map1, map2 map[string][]string) bool {
+	if len(map1) != len(map2) {
+		return false
+	}
+
+	for key, value1 := range map1 {
+		value2, exists := map2[key]
+		if !exists || !reflect.DeepEqual(value1, value2) {
```

Comment:
Hmm, would it better to just skip this step? Worst case even if the maps are equal, the new custom tags will just overwrite what's currently in the custom tags map?

Reply:
It all depends on the cost of having to perform the configuration update. In this case the cost is just locking a mutex to swap out the map, that's probably not a disruptive operation at the frequency this will occur. (In particular because we do optimize this by only calling the callback when something actually changed, even though we don't give only the diff to the callback function)

However - if you had to finish processing what you were processing, teardown the struct, and rebuild it, you'd probably want to optimize that by detecting the difference here.